components/src/core/DropdownToolbar/DropdownToolbar.tsx

```diff
@@ -0,0 +1,181 @@
+import React, { useState, useCallback, useRef } from 'react';
+import {
+    Typography,
+    Toolbar,
+    ListItemText,
+    Menu,
+    MenuItem,
+    ToolbarProps,
+    createStyles,
+    makeStyles,
+    useTheme,
+    Theme,
+} from '@material-ui/core';
+import { ArrowDropDown } from '@material-ui/icons';
+import clsx from 'clsx';
+
+const MENU_ITEM_HEIGHT = 48;
+
+export type ToolbarMenuItem = {
+    label: string;
+    onClick: () => void;
+};
+export type DropdownToolbarProps = ToolbarProps & {
+    title: string;
+    subtitleLabel: string | undefined;
+    menuItems?: ToolbarMenuItem[] | undefined;
+    navigationIcon?: JSX.Element;
+    customMenu?: JSX.Element;
+    classes?: DropdownToolbarClasses;
+};
+
+const useStyles = makeStyles((theme: Theme) =>
+    createStyles({
+        root: {
+            paddingLeft: theme.spacing(2),
+            paddingRight: theme.spacing(2),
+        },
+        navigationIconWrapper: {
+            marginRight: theme.spacing(4),
+            cursor: 'pointer',
+        },
+        title: {
+            lineHeight: 1,
+        },
+        subtitle: {
+            display: 'inline-flex',
+            cursor: 'pointer',
+        },
+        arrowDropdown: {
+            marginLeft: 4,
+        },
+        menuItem: {
+            height: MENU_ITEM_HEIGHT,
+        },
+    })
+);
+
+export type DropdownToolbarClasses = {
+    root?: string;
+    navigationIconWrapper?: string;
+    title?: string;
+    subtitle?: string;
+    arrowDropdown?: string;
+    menuItem?: string;
+};
+
+export const DropdownToolbar: React.FC<DropdownToolbarProps> = (props) => {
+    const { title, subtitleLabel, menuItems, navigationIcon, customMenu, style, classes } = props;
+    const [anchorEl, setAnchorEl] = useState(null);
+    const anchor = useRef(null);
+    const theme = useTheme();
+    const defaultClasses = useStyles(theme);
+    const getNavigationIcon = useCallback(() => {
+        if (navigationIcon) {
```

Comment:
Can we see about conditionally checking to see if the content passed in for navigationIcon is an icon or an IconButton? And apply the correct styles to get the right spacing for each case...It woud be nice if we don't have to depend on the user to correctly add in negative margins on their icon button.

Reply:
We can do this in a followup - let's make an enhancement issue for it.

Reply:
https://github.com/pxblue/react-component-library/issues/180